tasks/main.yml

```diff
@@ -21,36 +21,79 @@
     jboss_eap_artifact_url: "{{ jboss_eap_artifact_repo + '/' + jboss_eap_artifact_name }}"
     jboss_eap_artifact_dl_dest: "{{ '/tmp/' + jboss_eap_artifact_name }}"
     jboss_eap_library_dest: "/opt/{{ jboss_eap_user }}"
+    jboss_eap_patch_artifact_url: "{{ jboss_eap_artifact_repo + '/' + jboss_eap_patch_artifact_name }}"
+    jboss_eap_patch_dest: "/tmp/{{ jboss_eap_patch_artifact_name }}"
+  tags:
+    - jboss_eap
 
 - name: Set JBoss EAP Group Facts
   set_fact:
     jboss_eap_group: "{{ jboss_eap_user }}"
+  tags:
+    - jboss_eap
 
 - name: Set JBoss EAP Service Directories Facts
   set_fact:
     jboss_eap_home: "{{ jboss_eap_library_dest }}/jboss-eap-6.4"
     jboss_eap_service_conf_dir: "/etc/{{ jboss_eap_user }}"
     jboss_eap_service_log_dir: "/var/log/{{ jboss_eap_user }}"
     jboss_eap_service_data_dir: "/var/run/{{ jboss_eap_user }}"
+  tags:
+    - jboss_eap
 
 - name: Set JBoss EAP Service Files Facts
   set_fact:
     jboss_eap_service_conf_file: "{{ jboss_eap_service_conf_dir }}/{{ jboss_eap_user }}.conf"
     jboss_eap_runtime_conf_file: "{{ jboss_eap_home }}/bin/standalone.conf"
     jboss_eap_standalone_service_file: "/etc/systemd/system/{{ jboss_eap_user }}-standalone.service"
+  tags:
+    - jboss_eap
 
 #### Create Service Account
 
 - include: jboss_service.yml
 
-#### Download and unzip libraries
+#### Download and install base library and patch
 
-- name: Download JBoss EAP from Red Hat Customer Portal
+- name: Async Download JBoss EAP Patch from Red Hat Customer Portal
+  get_url:
+    url: "{{ jboss_eap_patch_artifact_url }}"
+    dest: "{{ jboss_eap_patch_dest }}"
+    url_username: "{{ rhn_username }}"
+    url_password: "{{ rhn_password }}"
+  async: 7200
+  poll: 0
+  register: jboss_eap_patch_download
+  tags:
+    - jboss_eap
+
+- name: Async Download JBoss EAP from Red Hat Customer Portal
   get_url:
     url: "{{ jboss_eap_artifact_url }}"
     dest: "{{ jboss_eap_artifact_dl_dest }}"
     url_username: "{{ rhn_username }}"
     url_password: "{{ rhn_password }}"
+  async: 7200
+  poll: 0
+  register: jboss_eap_download
+  tags:
+    - jboss_eap
+
+- name: 'Check On JBoss EAP Download Completion'
+  async_status: jid={{ jboss_eap_download.ansible_job_id }}
+  register: job_result1
+  until: job_result1.finished
+  retries: 5
+  delay: 10
+  tags:
+    - jboss_eap
+
+- name: 'Check On JBoss EAP Patch Download Completion'
+  async_status: jid={{ jboss_eap_patch_download.ansible_job_id }}
+  register: job_result2
+  until: job_result2.finished
+  retries: 5
+  delay: 10
```

Comment:
Yes that could be an issue. Should this be a variable? If so, what should the defaults be?